### PR TITLE
scripts: Check that LunarG / Valve Copyright is updated

### PR DESCRIPTION
**Pull request by juan-lunarg**
See https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/5142#discussion_r1083124269

### CODE REVIEW

**Review by ncesario-lunarg — APPROVED** [review]

File: scripts/check_code_format.py [hunk 1]
```diff
@@ -105,14 +105,15 @@ def VerifyCopyrights(commit, target_files):
...
-        copyright_match = re.search('Copyright .*(\d{4}) LunarG', open(file, encoding="utf-8", errors='ignore').read(1024))
-        if copyright_match:
-            copyright_year = copyright_match.group(1)
-            if int(commit_year) > int(copyright_year):
-                msg = 'Change written in {} but copyright ends in {}.'.format(commit_year, copyright_year)
-                CPrint('ERR_MSG', '\n' + file + ' has an out-of-date LunarG copyright notice. ' + msg)
-                retval = 1;
+        for company in ["LunarG", "Valve"]:
+            # Capture the last year on the line as a separate match. It should be the highest (or only year of the range)
+            copyright_match = re.search('Copyright .*(\d{4}) ' + company, open(file, encoding="utf-8", errors='ignore').read(1024))
```
Comment (lunarpapillo) [inline scripts/check_code_format.py:110]: Not your code, but this is pretty grisly.  It fails if the copyright isn't in the first 1K of the file, leaks an open file for every file it examines, breaks if two copyright lines for the same company appear in the file, regenerates `re` patterns repeatedly and needlessly, and relies on an obscure pattern match behavior (that `.` doesn't include a newline unless `re.M` is used).

Also, what should happen if no copyright is found in a given file?  I'd think that should be an error, but it isn't covered here.

I'll talk to you in person about a recommendation to clean this up (more code than I want to write in a web form).

**Review by lunarpapillo — CHANGES_REQUESTED** [review]
Bad script to start with... but your changes look good (at least no worse than the script around them).

I'd like to work with you to clean it up a bit.

I presume there's no Python standard for this project, and that `pylint` has never been run on this script...

File: scripts/check_code_format.py (hunk 1 above)
Comment (mark-lunarg) [reply scripts/check_code_format.py:110]: "Grisly" eh? [Writes in a little notebook with a very short pencil].

File: scripts/check_code_format.py [hunk 2]
```diff
@@ -105,14 +105,15 @@ def VerifyCopyrights(commit, target_files):
...
+            if copyright_match:
+                copyright_year = copyright_match.group(1)
+                if int(commit_year) > int(copyright_year):
+                    msg = 'Change written in {} but copyright ends in {}.'.format(commit_year, copyright_year)
```
[the remark this answers is not in the document]
Comment (ncesario-lunarg) [reply scripts/check_code_format.py:114]: We are using 3.7 as a minimum right, and I'll be bumping to 3.8 soon.

[the remark this answers is not in the document]
Comment (lunarpapillo) [reply scripts/check_code_format.py:114]: Please do a Python version check at the top of the script.  @Tony-LunarG already ran into confusion when he used an earlier Python.

With 3.7 using f-strings should be preferred over format strings.

**Pull request merged**